Fix type and add tests for "react-calendar-heatmap"

sboh1214 opened the pull request:
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [https://github.com/kevinsqi/react-calendar-heatmap](https://github.com/kevinsqi/react-calendar-heatmap)
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

DangerBotOSS commented:
<!--
  0 failure: 
  0 warning: 
  
  1 markdown notices
  DangerID: danger-id-Danger;
-->



Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## react-calendar-heatmap ([<kbd>unpkg</kbd>](https://unpkg.com/browse/react-calendar-heatmap@latest/))
was missing the following properties:
1. propTypes
1. defaultProps
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against df5f2c8914e8abda54f1c3fc51e710edbfbe63c6
</p>

jakebailey commented on types/react-calendar-heatmap/tsconfig.json:
```diff
@@ -16,6 +16,7 @@
         "types": [],
         "noEmit": true,
         "forceConsistentCasingInFileNames": true,
+        "esModuleInterop": true,
```
This config option should not be present in a types package, see: https://github.com/DefinitelyTyped/DefinitelyTyped#esmoduleinteropallowsyntheticdefaultimports

sboh1214 replied:
I removed it! Thanks

jakebailey reviewed (changes requested):
This seems like a complete change from the original code; can you just update the things you need updated without renaming and reordering all of the types, and changing the structure of the imports and exports?

See also the config change; that interop setting cannot be set in a DT package.

sboh1214 commented:
As your comments, I rollbacked reordering types, changing export structure.
However, I changed all unnecessary `Something | undefined` type to `Something`
And I changed to `export = something` expression because of error message.

jakebailey commented:
The `| undefined` are not unnecessary; see https://www.typescriptlang.org/tsconfig#exactOptionalPropertyTypes

sboh1214 commented:
@jakebailey Oh I didn't know that. I add new commit. Thanks!

jakebailey commented on types/react-calendar-heatmap/index.d.ts:
```diff
@@ -1,30 +1,42 @@
+// Type definitions for react-calendar-heatmap 1.8
 // Project: https://github.com/patientslikeme/react-calendar-heatmap
 // Definitions by: Keisuke Kan <https://github.com/9renpoto>
+//                 Seungbin Oh <https://github.com/sboh1214>
```
Are you intending to become an owner of this types package, or were you looking for where to add credit for your changes? If the latter, you shouldn't add your name here. https://github.com/DefinitelyTyped/DefinitelyTyped#definition-owners

jakebailey reviewed (changes requested):
Overall, I think you're changing too much in this PR... There was an exported Props type, but now it's got a different name and is no longer exported, so is a breaking change. I see you added `| undefined` back on, but looking at the original types, the `| undefined` was placed in the return type for many properties, which isn't correct.

Is there any way to distill your PR down to just the relevant changes so someone can review them? e.g., start with the original definitions, and apply just the fixes you need keeping the rest in place?
